systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesST.java

```diff
@@ -0,0 +1,112 @@
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static io.strimzi.systemtest.Constants.INTERNAL_CLIENTS_USED;
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Feature Gates should give us additional options on
+ * how to control and mature different behaviors in the operators.
+ * https://github.com/strimzi/proposals/blob/main/022-feature-gates.md
+ */
+@Tag(REGRESSION)
+public class FeatureGatesST extends AbstractST {
+
+    static final String NAMESPACE = "feature-gates-tests";
+    private static final Logger LOGGER = LogManager.getLogger(FeatureGatesST.class);
+
+    protected void deployClusterOperatorWithEnvVars(ExtensionContext extensionContext, List<EnvVar> envVars) {
+        prepareEnvForOperator(extensionContext, NAMESPACE, Arrays.asList(NAMESPACE, NAMESPACE));
+        applyBindings(extensionContext, NAMESPACE);
+        resourceManager.createResource(extensionContext, BundleResource.clusterOperator(NAMESPACE, envVars).build());
+    }
+
+    /**
+     * Control Plane Listener
+     * https://github.com/strimzi/proposals/blob/main/025-control-plain-listener.md
+     */
+    @IsolatedTest("Feature Gates test for ControlPlainListener")
+    @Tag(INTERNAL_CLIENTS_USED)
+    public void testControlPlaneListenerFeatureGate(ExtensionContext extensionContext) throws InterruptedException {
+        final String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+        final String producerName = "producer-test";
+        final String consumerName = "consumer-test";
+        final String topicName = "topic-fg-cpl";
+        int sendingTime = 300;
+        List<EnvVar> testEnvVars = new ArrayList<>();
+        int kafkaReplicas = 2;
+
+        testEnvVars.add(new EnvVar(Environment.STRIMZI_FEATURE_GATES_ENV, "+ControlPlaneListener", null));
+        deployClusterOperatorWithEnvVars(extensionContext, testEnvVars);
+
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaPersistent(clusterName, kafkaReplicas).build());
+
+        LOGGER.info("Check for presence of ContainerPort 9090/tcp (tcp-ctrlplane) in first Kafka pod.");
+        final Pod kafkaPod = PodUtils.getPodsByPrefixInNameWithDynamicWait(NAMESPACE, clusterName + "-kafka-").get(0);
+        ContainerPort expectedControlPlaneContainerPort = new ContainerPort(9090, null, null, "tcp-ctrlplane", "TCP");
+        List<ContainerPort> kafkaPodPorts = kafkaPod.getSpec().getContainers().get(0).getPorts();
+        assertTrue(kafkaPodPorts.contains(expectedControlPlaneContainerPort));
+
+        LOGGER.info("Try to send some messages to Kafka over the next {} seconds.", sendingTime);
+        // use example clients for jobs - sending messages for 3-5mins
+        KafkaBasicExampleClients kafkaBasicClientJob = new KafkaBasicExampleClients.Builder()
```

Comment:
Where do we create the topic used by this? Can we make sure it has multiple replicas?

Reply:
auto-creation is used I guess

Reply:
Yes, I will create it before with replication=2